system/inc/system_sleep.h

```diff
@@ -41,12 +51,406 @@ typedef enum System_Sleep_Flag
     SYSTEM_SLEEP_FLAG_NO_WAIT = 0x04
 } System_Sleep_Flag;
 
+
+/**
+ * Since sleep 2.0
+ */
+enum class SystemSleepMode: uint8_t {
+    NONE            = HAL_SLEEP_MODE_NONE,
+    STOP            = HAL_SLEEP_MODE_STOP,
+    ULTRA_LOW_POWER = HAL_SLEEP_MODE_ULTRA_LOW_POWER,
+    HIBERNATE       = HAL_SLEEP_MODE_HIBERNATE,
+};
+
+enum class SystemSleepWait: uint8_t {
+    NO_WAIT = HAL_SLEEP_WAIT_NO_WAIT,
+    CLOUD   = HAL_SLEEP_WAIT_CLOUD
+};
+
+enum class SystemSleepWakeupReason: uint16_t {
+    UNKNOWN = HAL_WAKEUP_SOURCE_TYPE_UNKNOWN,
+    BY_GPIO = HAL_WAKEUP_SOURCE_TYPE_GPIO,
```

Comment:
For Gen2 platforms, `ADC`, ` GPIO` etc. are also macros in chip level header file like stm32f2xx.h. That would result compilation errors. Any better naming scheme? Neither I like this style honestly.

Reply:
Ugh. Does it help if we change the order of the includes?

Reply:
It won't help as it anyway includes stm32f2xx.h eventually previous to the code where `ADC` is presented in user application.

Reply:
So, I'll keep them as-is?

Reply:
It still somewhat annoys me that we can't name things how we want. Let's leave things as-is for now, but I'm anyway going to take a look today to see if we can fix it somehow :smile:

Reply:
As a compromise, we can potentially use Camel-Case for enum values to avoid the conflict with macro definitions, as we might still suffer from this conflict in the future.